### ISSUE TITLE
[Shared] Improve `ObserveAttributesMixin`

### ISSUE THREAD

**Issue by anna-lach** [sl-design-system/components#3101]
### Provide a general summary of the issue here

Improve `ObserveAttributesMixin` to get it working with not only as it works now, so:
- When there is an attribute applied on the host element it's moved to the target (focusable) element and removed from the host element,

But it should also be possible to get it working this way:
- When there is an attribute applied on the host element and we want to add it to the target (focusable) element as well and keep both (on host and target element). Probably it can be achieved by having a separate array for such attributes applied to the `ObserveAttributesMixin`.

We need to check which attributes we want to keep on both host and focusable (internal) element. Let's discuss this one with @a11ymiko  first. 

First attribute we know it should be set on host and focusable element is `aria-disabled`. Once the change in the `ObserveAttributesMixin` is ready, please refactor (within this issue) setting `aria-disabled` in the `menu-button` and also in other places where it's necessary. 
Please also make some changes to other components which are using other attributes we want to keep on both host and focusable element (copy of it).

### 👤 Your name

SLDS Team

### 🧢 Your Product/Team

SLDS
